Code/Framework/AzFramework/AzFramework/Device/DeviceAttributeInterface.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/std/any.h>
+
+namespace AzFramework
+{
+    // DeviceAttribute provides an interface for requesting information about a
+    // single device attribute like model, GPU type, RAM amount etc.
+    class DeviceAttribute
+    {
+    public:
+        virtual ~DeviceAttribute() = default;
+
+        //! Get the name of the device attribute e.g. gpuMemory, gpuVendor, customAttribute42
+        virtual AZStd::string_view GetName() const = 0;
+
+        //! Get a description of this device attribute, used for help text and eventual UI
+        virtual AZStd::string_view GetDescription() const = 0;
+
+        //! Evaluate a rule and return true if there is a match for this device attribute
+        virtual bool Evaluate(AZStd::string_view rule) const = 0;
+
+        //! Get the value of this attribute
+        virtual AZStd::any GetValue() const = 0;
+    };
+
+    // DeviceAttributeRegistrarInterface provides an interface for registering and accessing
+    // DeviceAttributes which describes a single device attribute.
+    class DeviceAttributeRegistrarInterface
+    {
+    public:
+        AZ_RTTI(DeviceAttributeRegistrarInterface, "{D6B65DF8-8275-42F8-B84D-4F9ACBECC7C2}");
+        virtual ~DeviceAttributeRegistrarInterface() = default;
+
+        //! Register a device attribute interface, deviceAttribute must be unique, returns true on success
+        virtual bool RegisterDeviceAttribute(AZStd::unique_ptr<DeviceAttribute> deviceAttributeInterface) = 0;

```

Comment:
We should add an UnregisterDeviceAttribute function for completeness.
```suggestion
        //! Register a device attribute interface, deviceAttribute must be unique, returns true on success
        virtual bool RegisterDeviceAttribute(AZStd::unique_ptr<DeviceAttribute> deviceAttributeInterface) = 0;
        
        //! Unregister a device attribute interface, returns true if a device with the specified name was registered
        virtual bool UnregisterDeviceAttribute(AZStd::string_view deviceAttributeName) = 0;

```